custom_components/aqara_gateway/core/gateway.py

```diff
@@ -483,7 +483,6 @@ def on_connect(self, client, userdata, flags, ret):
     def on_disconnect(self, client, userdata, ret):
         # pylint: disable=unused-argument
         """ on disconnect to mqtt server """
-        self._mqttc.disconnect()
```

Comment:
Calling `_mqttc.disconnect()` will cause the mqtt loop thread to exit which breaks the internal reconnecting mechanism of `_mqttc.loop_start()`.